### PR TITLE
Set local path hint to remote repo name

### PR DESCRIPTION
**Pull request by bigdavedev**
References issue #152 

Signed-off-by: Dave Brown <d.brown@bigdavedev.com>

### CODE REVIEW

File: src/me/sheimi/sgit/dialogs/CloneDialog.java [hunk 1]
```diff
@@ -115,19 +129,25 @@ public void onClick(View view) {
...
+                return;
+            }
         if (localPath.contains("/")) {
             showToastMessage(R.string.alert_localpath_format);
             mLocalPath.setError(getString(R.string.alert_localpath_format));
             mLocalPath.requestFocus();
             return;
         }
 
+        // If user is accepting the default path in the hint, we need to set localPath to
```
Comment (bigdavedev) [inline src/me/sheimi/sgit/dialogs/CloneDialog.java]: This block could probably be moved to the initialisation point of ```localPath``` and thus negate the need for the added check on ```mLocalPath``` in the previous hunk.  I will fix that later tonight and ensure the behaviour is correct.

**Comment by bigdavedev** [comment]
Some tidying up left to do.  Will update this PR tonight.

File: src/me/sheimi/sgit/dialogs/CloneDialog.java [hunk 2]
```diff
@@ -64,6 +64,20 @@ public Dialog onCreateDialog(Bundle savedInstanceState) {
             mLocalPath.setText(Repo.TEST_LOCAL);
         }
 
+        mRemoteURL.setOnFocusChangeListener(new View.OnFocusChangeListener() {
+            @Override
+            public void onFocusChange(View view, boolean hasFocus) {
+                if (!hasFocus) {
+                    final String remoteUrl = mRemoteURL.getText().toString();
+                    final int lastSlash = remoteUrl.lastIndexOf("/");
+                    final String localHint = remoteUrl.substring(lastSlash + 1);
```
Comment (maks) [inline src/me/sheimi/sgit/dialogs/CloneDialog.java]: @bigdavedev I think it may also be a good idea to strip any ".git" suffix too.

**Comment by maks** [comment]
@bigdavedev thanks very much for the PR! Apart from 1 tiny quibble looks good to me.

File: src/me/sheimi/sgit/dialogs/CloneDialog.java (hunk 2 above)
Comment (bigdavedev) [inline src/me/sheimi/sgit/dialogs/CloneDialog.java]: I agree.  Even though git handles this situation itself, it is certainly more visually appealing if it is never there to begin with.

**Pull request merged**